pxr/usdImaging/usdviewq/stageView.py

```diff
@@ -628,20 +639,18 @@ def draw(self, qglwidget):
 
         for name in self._groups:
             group = self._groups[name]
-
-            tex = qglwidget.bindTexture(group.qimage, GL.GL_TEXTURE_2D, GL.GL_RGBA,
-                                        QtOpenGL.QGLContext.NoBindOption)
-            GL.glUniform4f(self._glslProgram.uniformLocations["rect"],
-                           2*group.x/width - 1,
-                           1 - 2*group.y/height - 2*group.h/height,
-                           2*group.w/width,
-                           2*group.h/height)
-            GL.glUniform1i(self._glslProgram.uniformLocations["tex"], 0)
-            GL.glActiveTexture(GL.GL_TEXTURE0)
-            GL.glBindTexture(GL.GL_TEXTURE_2D, tex)
-            GL.glDrawArrays(GL.GL_TRIANGLE_STRIP, 0, 4)
-
-            GL.glDeleteTextures(tex)
+            if PySideModule == 'PySide6':
+                tex = QtOpenGL.QOpenGLTexture(group.qimage)
+                tex.bind()
+                self.drawWidgetTexture(width, height, group)
+                tex.release()
+                tex.destroy()
+            else:
+                tex = qglwidget.bindTexture(group.qimage, GL.GL_TEXTURE_2D, GL.GL_RGBA,
+                                            QtOpenGL.QGLContext.NoBindOption)
+                GL.glBindTexture(GL.GL_TEXTURE_2D, tex)
+                self.drawWidgetTexture(width, height, group)
+                GL.glDeleteTextures(tex)
```

Comment:
@jonny-apple , we'd really like to keep all Qt/pyside version logic in qt.py.  I think this one could be quite nicely wrapped up in a BindTexture context object implemented appropriately for the two different versions, maybe injected into QtOpenGL ?

Reply:
Good point Spiff, I'll move the logic back to qt.py.

Reply:
I've injected a BindTexture and ReleaseTexture to the QGLWidget class.